src/xrpld/app/misc/detail/ValidatorList.cpp

```diff
@@ -1749,11 +1779,38 @@ ValidatorList::calculateQuorum(
         return *minimumQuorum_;
     }
 
-    // Do not use achievable quorum until lists from all configured
-    // publishers are available
-    for (auto const& list : publisherLists_)
+    if (!publisherLists_.empty())
     {
-        if (list.second.status != PublisherStatus::available)
+        // Do not use achievable quorum until lists from a sufficient number of
+        // configured publishers are available
+        std::size_t unavailable = 0;
+        for (auto const& list : publisherLists_)
+        {
+            if (list.second.status != PublisherStatus::available)
+                unavailable += 1;
+        }
+        // There are two, subtly different, sides to list threshold
+        //
+        // 1. The minimum required intersection between lists listThreshold_
+        //    for a validator to be included in trustedMasterKeys_.
+        //    If this many (or more) publishers are unavailable, we are likely
+        //    to NOT include a validator which otherwise would have been used.
+        // 2. The minimum number of lists which, when UNAVAILABLE, will prevent
+        //    us from hitting the above threshold on ANY validator. This is
+        //    calculated as:
+        //      N - M + 1
```

Comment:
To aid in my review, can you provide a bit more detail on this view @Bronek ? I'm not following the distinction between case 1 and case 2 here. I think case #2 is saying that until we have at least the `listThreshold_` number of distinct* published lists available, then nothing can hit the threshold because there is simply not enough lists.

I'm not clear on #2. Is there an example you could work through in a GH comment?

* - distinct means having a different publisher key versus a distinct URL/domain.

Reply:
I will welcome suggestions how to formulate this better !

1. refers to the situation when, out of N configured lists, we are missing M or more. In this situation we will not be using validators which these lists might have provided us with. This retains the original intent of this section, when 1 was logically a hardcoded "errorThreshold" (an "intersection of 1" is just an union)

2. refers to situation - as you correctly recognized - when we are missing so many lists that it is impossible for any validator to meed the required threshold M (any, except _self_). This prevents an occurrence when we would calculate a quorum of 1, making this _self_ validator its own 100% quorum. In the original logic it is not possible, and I wanted to avoid introducing such behaviour here. Also see unit test change in commit https://github.com/XRPLF/rippled/pull/5112/commits/57b451cab070f696ef72ba1a1d9a1e2604f991b3

Reply:
Got it. I think this formulation works. The only alternative I can see is framing it in terms of available rather than unavailable number of lists, but thats not so different. 

On second read, I believe the "Ideally we want both to be equal ..." bit is what threw me off the most; maybe just drop that? But that's just me.

Reply:
I updated this comment, hopefully it is clearer now.

Reply:
That change does help. 👍